head.S

```diff
@@ -61,8 +60,7 @@ GLOBAL(_entry)
 
 	/* Check if we have enough space available for stack. */
 	lea	bootloader_data(%ebp), %ebx
-	/* TODO: below uses bootloader_data.size, must be changed for SLRT. */
-	movzwl	%ss:2(%ebx), %edx
+	movzwl	%ss:8(%ebx), %edx       /* bootloader_data.size */
```

Comment:
Not that this will have any effect given 64KiB size limit, but unlike size of tags, size of SLRT is 4 bytes, so instruction could be changed to read 4 bytes as well.

Reply:
`end_of_slrt()` assumes that whole bootloader data is within SLB, so it should be checked here.

https://github.com/TrenchBoot/secure-kernel-loader/compare/f2a18cb10d6548d177ca1eff9b8334344b2a306b..b9aaa5c37b91adeed72a067dde349129b9970fde#diff-8294dbaee32a9d34075481810244e65d4b352c8dc16a89c4a37fe7de4f4390b1R63